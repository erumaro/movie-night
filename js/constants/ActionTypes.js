export const FETCH_POPULAR_MOVIES = "FETCH_POPULAR_MOVIES";
export const FETCH_UPCOMING_MOVIES = "FETCH_UPCOMING_MOVIES";
export const FETCH_PLAYING_MOVIES = "FETCH_PLAYING_MOVIES";
export const FETCH_TOPRATED_MOVIES = "FETCH_TOPRATED_MOVIES";
export const FETCH_MOVIE_DETAILS = "FETCH_MOVIE_DETAILS";
export const FETCH_SEARCH_RESULTS = "FETCH_SEARCH_RESULTS";