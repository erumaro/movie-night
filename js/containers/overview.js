// React
import react {component} from "react";

  