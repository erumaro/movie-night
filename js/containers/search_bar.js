// React
import React, { Component } from "react";

class SearchBar extends Component {
	constructor(props) {
		super(props);
		this.state = {
			term: ""
		};
	}
	handleChange(e) {
		let selected = e.target.id;
		console.log(selected)
	}
	render(){
		return (
			<div id="search-bar">
				<form>
					<span>TV</span>
				  <input 
				  	onChange={e => this.handleChange(e)}
				  	id="tv" 
				  	type="radio" 
				  	name="select" 
				  	value="tv"
				  	defaultChecked
				  	 />
				  	 <span>MOVIES</span>
				  <input 
					  onChange={e => this.handleChange(e)} 
					  id="movie" 
					  type="radio" 
					  name="select" 
					  value="movies" />
				</form>
				<input 
					value={this.state.term}
					onChange={e => this.onInputChange(e.target.value)} />
				<div id="search-results">
				</div>
			</div>
		);
	}
	onInputChange(term) {
		this.setState({term});
		//this.props.onSearchTermChange(term);
		console.log(this.state.term);
	}

}


export default SearchBar;



