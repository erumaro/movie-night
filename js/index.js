// StyleSheets
require("../css/00-global.css");
require("../css/01-movieoverview.css");

// React
import React from 'react';
import ReactDOM from 'react-dom';

// Redux
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import ReduxPromise from "redux-promise";
import reducers from './reducers';

// Redux Logger
import createLogger from 'redux-logger';

// React Router
import {Router, Route, IndexRoute, Link, IndexLink, IndexRedirect, hashHistory} from 'react-router';

// Components / Containers
import App from './components/app.js'
import MovieOverview from './containers/movie_overview.js'
import TvOverview from './containers/tv_overview.js'
import MovieDetails from './containers/movie_details.js'

// Consts
const logger = createLogger();
const createStoreWithMiddleware = applyMiddleware(ReduxPromise, logger)(createStore);

const app = document.getElementById("app");

ReactDOM.render(
  <Provider store={createStoreWithMiddleware(reducers)}>
		<App />
	</Provider>,
app

);