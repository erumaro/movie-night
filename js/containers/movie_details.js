// React
import React, { Component } from "react";

// Redux
import { connect } from "react-redux";
import { bindActionCreators } from "redux";

// Actions
import { fetchMovieDetails } from "../actions/index.js";


// Containers/Components
import MovieOverview from "./movie_overview";
// React Router
import {Router, Route, Link, hashHistory } from 'react-router';


class MovieDetails extends Component {
	
	componentWillMount() {
		// this.props.fetchPopularMovies();
		// this.props.fetchUpcomingMovies();
		// this.props.fetchNowPlayingMovies();
		//this.props.fetchTopRatedMovies();
		this.props.fetchMovieDetails(this.props.params.id);

	}

	renderMovieDetails(){

			
			if(this.props.movieDetails[0]){
				let releaseDate = this.props.movieDetails[0].release_date;
				let genres = "";
				let i = 0;
				let year = this.props.movieDetails[0].release_date.substring(0,4);
				let genreArr =  this.props.movieDetails[0].genres;
				let genre = genreArr.forEach((genre) => {
					i !== genreArr.length - 1 ? genres += (genre.name + ", ") : genres += genre.name;
					i++;
				}) 
			return(
			<div id="detail-wrapper">
				<div id="detail-container">
					<figure>
						<img src={"https://image.tmdb.org/t/p/w500/" + this.props.movieDetails[0].poster_path}  alt="poster"/>
					</figure>
					<section>
						<h1>{this.props.movieDetails[0].title + " (" + year + ")"}</h1>
						<span>{genres}</span>
						<h2>Overview</h2>
						<p>
							{this.props.movieDetails[0].overview}
						</p>
						<h2>Facts for geeks</h2>
						<p className="facts">Release: {releaseDate}</p>
						<p className="facts">Vote Avg: {this.props.movieDetails[0].vote_average}</p>
						<p className="facts">Vote Count: {this.props.movieDetails[0].vote_count}</p>
					</section>
				</div>	
			</div>
			)
		}
		else {
			<div>Loading movies...</div>
		}
	}

	render(){

		 if(this.props.movieDetails[0]){
			console.log(this.props.movieDetails[0].title)
			} 
	
		return (
			<div className="fullPage">
				<h1>Header Title</h1>
				<div className="movieContainer">
					{this.renderMovieDetails()}
				</div>
			</div>
		)
	}

}


function mapStateToProps(state){
	return { movieDetails: state.movieDetails };
}

function mapDispatchToProps(dispatch){
	return bindActionCreators ({fetchMovieDetails}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(MovieDetails);