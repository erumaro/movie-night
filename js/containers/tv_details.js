// React
import React, { Component } from "react";

// Redux
import { connect } from "react-redux";
import { bindActionCreators } from "redux";

// Actions
import { fetchMovieDetails } from "../actions/index.js";


// Containers/Components
import TvOverview from "./tv_overview";

// React Router
import {Router, Route, Link, hashHistory } from 'react-router';


class TvDetails extends Component {
	
	componentWillMount() {
		// this.props.fetchPopularMovies();
		// this.props.fetchUpcomingMovies();
		// this.props.fetchNowPlayingMovies();
		//this.props.fetchTopRatedMovies();
		this.props.fetchTvDetails(this.props.params.tvId);

	}

	renderMovieDetails(){
	
			return(
			
			<div>Tv Details</div>
	
			)
	}

	render(){

		 if(this.props.tvDetails[0]){
			console.log(this.props.tvDetails[0].title)
		} 
	
		return (
			<div className="fullPage">
				<h1>Header Title</h1>
				
				<div className="movieContainer">
					{this.renderTvDetails.bind(this)}
				</div>
	
			</div>
		)
	}

}


function mapStateToProps(state){
	return { movieDetails: state.tvDetails };
}

function mapDispatchToProps(dispatch){
	return bindActionCreators ({fetchTvDetails}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(TvDetails);