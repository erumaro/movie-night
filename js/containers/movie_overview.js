// React
import React, { Component } from "react";

// Redux
import { connect } from "react-redux";
import { bindActionCreators } from "redux";

// Actions
import { fetchPopularMovies } from "../actions/index.js";
import { fetchUpcomingMovies } from "../actions/index.js";
import { fetchNowPlayingMovies } from "../actions/index.js";
import { fetchTopRatedMovies } from "../actions/index.js";

// Containers/Components
import { MovieDetails } from "./movie_details.js"
import SearchBar from './search_bar.js';

// React Router
import {Router, Route, Link, hashHistory } from 'react-router';

// let category = this.props.params.menuCategory

class MovieOverview extends Component {
	
	componentWillMount() {
		// this.props.fetchPopularMovies();
		// this.props.fetchUpcomingMovies();
		// this.props.fetchNowPlayingMovies();
		this.props.fetchTopRatedMovies();
		
		// Hämta "Category" från Router test
		// this.props.category();
	}

	renderMovies(movieList){
	
			return(
			
			<article key={Math.random()} className="movieThumb">
			
				{movieList.backdrop_path === null ? <img src="../img/no_image.png"></img> : <img src={"https://image.tmdb.org/t/p/w500_and_h281_bestv2/" + movieList.backdrop_path}></img>}

				<section className="info">
					
					<section className="topInfo">
						<Link to={`/movieDetails/${movieList.id}`}><h2>{movieList.title}</h2></Link>
						<h3>{movieList.release_date}</h3>
					</section>
					
					<section className="bottomInfo">
						<h3>#Votes: {movieList.vote_count}</h3>
						<h3>Average vote: {movieList.vote_average}</h3>
					</section>
	
				</section>
			</article>
	
			)
	}

	render(){

		/* if(this.props.movies[0]){
			console.log("https://image.tmdb.org/t/p/w500_and_h281_bestv2/" + this.props.movies[1].backdrop_path)
		} */
	
		return (
			<div className="fullPage centeringDiv">
				<SearchBar />
				<h1>Header Title</h1>
				
				<div className="movieContainer">
					{this.props.movies.map(this.renderMovies)}
				</div>
	
			</div>
		)
	}

}


function mapStateToProps(state){
	return { movies: state.movies };
}

function mapDispatchToProps(dispatch){
	return bindActionCreators ({fetchPopularMovies, fetchUpcomingMovies, fetchNowPlayingMovies, fetchTopRatedMovies}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(MovieOverview);



//  asjdkasljd </Link>