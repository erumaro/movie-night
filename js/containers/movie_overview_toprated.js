// React
import React, { Component } from "react";

// Redux
import { connect } from "react-redux";
import { bindActionCreators } from "redux";

// Actions
import { fetchTopRatedMovies } from "../actions/index.js";

// Containers/Components
import { MovieDetails } from "./movie_details.js"

// React Router
import {Router, Route, Link, hashHistory } from 'react-router';


class MovieOverviewToprated extends Component {

	componentWillMount() {
		this.props.fetchTopRatedMovies();
	}
	

	renderMovies(movieList){
	
			return(
			
			<article key={Math.random()} className="movieThumb">
			
				{movieList.backdrop_path === null ? <img src="../img/no_image.png"></img> : <img src={"https://image.tmdb.org/t/p/w500_and_h281_bestv2/" + movieList.backdrop_path}></img>}

				<section className="info">
					
					<section className="topInfo">
						<Link to={`/movieDetails/${movieList.id}`}><h2>{movieList.title}</h2></Link>
						<h3>{movieList.release_date}</h3>
					</section>
					
					<section className="bottomInfo">
						<h3>#Votes: {movieList.vote_count}</h3>
						<h3>Average vote: {movieList.vote_average}</h3>
					</section>
	
				</section>
			</article>
	
			)
	}

	render(){

		return (
			<div className="fullPage centeringDiv">
				<h1>Top Rated Movies</h1>
				
				<div className="movieContainer">
					{this.props.movies.map(this.renderMovies)}
				</div>
	
			</div>
		)
	}

}


function mapStateToProps(state){
	return { movies: state.movies };
}

function mapDispatchToProps(dispatch){
	return bindActionCreators ({fetchTopRatedMovies}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(MovieOverviewToprated);