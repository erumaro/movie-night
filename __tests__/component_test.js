import React from 'react';
import renderer from 'react-test-renderer';
import { fromJS } from 'immutable';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MovieDetails } from '../js/containers/movie_details';
import configureMockStore from 'redux-mock-store';
import axios from 'axios';
import * as types from "../js/constants/ActionTypes.js";

describe('components', function() {
  describe('<MovieDetails />', function() {

    it('Link renders correctly', () => {
  const tree = renderer.create(
    <Link page="http://www.facebook.com">Facebook</Link>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

  });
});

