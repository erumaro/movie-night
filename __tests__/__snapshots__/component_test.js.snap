exports[`Components <SearchBarTv /> renders correctly 1`] = `
<div
  id="search-bar-wrapper">
  <input
    onChange={[Function]}
    placeholder="Search Tv"
    type="text" />
</div>
`;

exports[`Components <TvDetails /> renders correctly 1`] = `
<div
  className="fullPage">
  <div
    className="movieContainer" />
</div>
`;

exports[`Components <TvOverviewPlaying /> renders correctly 1`] = `
<div>
  <div
    id="search-bar-wrapper">
    <input
      onChange={[Function]}
      placeholder="Search Tv"
      type="text" />
  </div>
  <div
    className="fullPage centeringDiv">
    <h1>
      TV series Airing today
    </h1>
    <div
      className="movieContainer" />
  </div>
</div>
`;

exports[`Components <TvOverviewPopular /> renders correctly 1`] = `
<div>
  <div
    id="search-bar-wrapper">
    <input
      onChange={[Function]}
      placeholder="Search Tv"
      type="text" />
  </div>
  <div
    className="fullPage centeringDiv">
    <h1>
      Popular TV series
    </h1>
    <div
      className="movieContainer" />
  </div>
</div>
`;

exports[`Components <TvOverviewToprated /> renders correctly 1`] = `
<div>
  <div
    id="search-bar-wrapper">
    <input
      onChange={[Function]}
      placeholder="Search Tv"
      type="text" />
  </div>
  <div
    className="fullPage centeringDiv">
    <h1>
      Top Rated TV series
    </h1>
    <div
      className="movieContainer" />
  </div>
</div>
`;

exports[`Components <TvOverviewUpcoming /> renders correctly 1`] = `
<div>
  <div
    id="search-bar-wrapper">
    <input
      onChange={[Function]}
      placeholder="Search Tv"
      type="text" />
  </div>
  <div
    className="fullPage centeringDiv">
    <h1>
      Upcoming TV series
    </h1>
    <div
      className="movieContainer" />
  </div>
</div>
`;

exports[`Components Movie Details should render correctly 1`] = `
<div
  className="fullPage">
  <div
    className="movieContainer" />
</div>
`;

exports[`Components Movies Playing should render correctly 1`] = `
<div>
  <div
    className="fullPage centeringDiv">
    <h1>
      Movies Now Playing
    </h1>
    <div
      id="search-bar-wrapper">
      <input
        onChange={[Function]}
        placeholder="Search Movies"
        type="text" />
    </div>
    <div
      className="movieContainer" />
  </div>
</div>
`;

exports[`Components Movies Popular should render correctly 1`] = `
<div>
  <div
    id="search-bar-wrapper">
    <input
      onChange={[Function]}
      placeholder="Search Movies"
      type="text" />
  </div>
  <div
    className="fullPage centeringDiv">
    <h1>
      Popular Movies
    </h1>
    <div
      className="movieContainer" />
  </div>
</div>
`;

exports[`Components Movies Toprated should render correctly 1`] = `
<div>
  <div
    id="search-bar-wrapper">
    <input
      onChange={[Function]}
      placeholder="Search Movies"
      type="text" />
  </div>
  <div
    className="fullPage centeringDiv">
    <h1>
      Top Rated Movies
    </h1>
    <div
      className="movieContainer" />
  </div>
</div>
`;

exports[`Components Movies Upcoming should render correctly 1`] = `
<div>
  <div
    id="search-bar-wrapper">
    <input
      onChange={[Function]}
      placeholder="Search Movies"
      type="text" />
  </div>
  <div
    className="fullPage centeringDiv">
    <h1>
      Upcoming Movies this month
    </h1>
    <div
      className="movieContainer" />
  </div>
</div>
`;

exports[`Components Searchbar should render correctly 1`] = `
<div
  id="search-bar-wrapper">
  <input
    onChange={[Function]}
    placeholder="Search Movies"
    type="text" />
</div>
`;
